server/src/test/java/org/elasticsearch/index/codec/postings/ES812PostingsFormatTests.java

```diff
@@ -0,0 +1,152 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Modifications copyright (C) 2022 Elasticsearch B.V.
+ */
+package org.elasticsearch.index.codec.postings;
+
+import org.apache.lucene.codecs.Codec;
+import org.apache.lucene.codecs.CompetitiveImpactAccumulator;
+import org.apache.lucene.codecs.lucene90.blocktree.FieldReader;
+import org.apache.lucene.codecs.lucene90.blocktree.Stats;
+import org.apache.lucene.codecs.lucene99.Lucene99PostingsFormat;
+import org.apache.lucene.codecs.lucene99.Lucene99SkipWriter;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.Impact;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.store.ByteArrayDataInput;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.apache.lucene.tests.analysis.MockAnalyzer;
+import org.apache.lucene.tests.index.BasePostingsFormatTestCase;
+import org.apache.lucene.tests.util.TestUtil;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class ES812PostingsFormatTests extends BasePostingsFormatTestCase {
+    private final Codec codec = TestUtil.alwaysPostingsFormat(new ES812PostingsFormat());
+
+    @Override
+    protected Codec getCodec() {
+        return codec;
+    }
+
+    /** Make sure the final sub-block(s) are not skipped. */
+    public void testFinalBlock() throws Exception {
+        Directory d = newDirectory();
+        IndexWriter w = new IndexWriter(d, new IndexWriterConfig(new MockAnalyzer(random())));
+        for (int i = 0; i < 25; i++) {
+            Document doc = new Document();
+            doc.add(newStringField("field", Character.toString((char) (97 + i)), Field.Store.NO));
+            doc.add(newStringField("field", "z" + Character.toString((char) (97 + i)), Field.Store.NO));
+            w.addDocument(doc);
+        }
+        w.forceMerge(1);
+
+        DirectoryReader r = DirectoryReader.open(w);
+        assertEquals(1, r.leaves().size());
+        FieldReader field = (FieldReader) r.leaves().get(0).reader().terms("field");
+        // We should see exactly two blocks: one root block (prefix empty string) and one block for z*
+        // terms (prefix z):
+        Stats stats = field.getStats();
+        assertEquals(0, stats.floorBlockCount);
+        assertEquals(2, stats.nonFloorBlockCount);
+        r.close();
+        w.close();
+        d.close();
+    }
+
+    private void shouldFail(int minItemsInBlock, int maxItemsInBlock) {
+        expectThrows(IllegalArgumentException.class, () -> { new Lucene99PostingsFormat(minItemsInBlock, maxItemsInBlock); });
```

Comment:
Should we replace it with the ES format?

```suggestion
        expectThrows(IllegalArgumentException.class, () -> { new ES812PostingsFormat(minItemsInBlock, maxItemsInBlock); });
```

Reply:
I remove the test, let me know if you think we should allow configuring the items size.

Reply:
Removing the test is good! :ship: